backend/index.js

```diff
@@ -0,0 +1,35 @@
+import fs from "fs";
+import express from "express";
+
+const app = express();
+
+let sessionCounter = 0;
+let counter;
+
+try {
+  fs.existsSync("temp/my-counter-visited-times.txt")
+    ? (counter = fs.readFileSync("temp/my-counter-visited-times.txt", "utf8"))
+    : (counter = 0);
+} catch (e) {
+  console.log(e);
+}
+
+app.get("/", (req, res) => {
+  sessionCounter++;
+  counter = parseInt(counter);
+  res.send(
+    `This page was visited ${
+      counter + sessionCounter
+    } times in total, ${sessionCounter} out of which was during the current code execution session ( statuscode:${
+      res.statusCode
+    } ) `
+  );
+
+  let str = JSON.stringify(counter + sessionCounter);
+  fs.writeFile("temp/my-counter-visited-times.txt", str, (err) => {
+    if (err) throw err;
+    console.log("The file has been saved!");
+  });
```

Comment:
Callbacks are bobo. Use try catch instead :)

Reply:
I think starting from node 12+ you can also use fs/promises which allows you to use async/await with `fs` utils.
e.g.
```
import { writeFile } from 'fs/promises';
*somewhere where you can use await*
try {
  await writeFile(....);
catch ...
```

But to be honest I'd simply use writeFileSync instead as here you don't have any problems with the size of the file or things like that.